src/BaseHostPlugin.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Allow to modify the the schema and the host of an existing request.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class BaseHostPlugin implements Plugin
+{
+    /**
+     * @var UriInterface
+     */
+    private $baseUri;
+
+    /**
+     * @var bool
+     */
+    private $replace;
+
+    /**
+     * @param UriInterface $baseUri
```

Comment:
from a validation point of view it makes sense to use an object here, but we should call it $host and say that only the host part of the uri is used.